magnet_coils.py

```diff
@@ -0,0 +1,350 @@
+import cubit
+import numpy as np
+import logging
+
+
+# Get logger defined in parametric_stellarator.py
+logger = logging.getLogger('log')
+
+
+def orient_rectangle(path_origin, surf_id, t_vec, norm, rot_axis, rot_ang_norm):
+    """Orients rectangular cross-section in the normal plane such that its
+    thickness direction faces the origin.
+    
+    Arguments:
+        path_origin (int): index of initial point in filament path.
+        surf_id (int): index of cross-section surface.
+        t_vec (list of float): cross-section thickness vector.
+        norm (list of float): cross-section normal vector.
+        rot_axis (list of float): axis about which to rotate the cross-section.
+        rot_ang_norm (float): angle by which cross-section was rotated to align
+            its normal with the initial point tangent (deg).
+    """
+    # Determine orientation of thickness vector after cross-section was
+    # oriented along filament origin tangent
+
+    # Compute part of thickness vector parallel to rotation axis
+    t_vec_par = np.inner(t_vec, rot_axis)*rot_axis
+    t_vec_par = t_vec_par/np.linalg.norm(t_vec_par)
+    # Compute part of thickness vector orthogonal to rotation axis
+    t_vec_perp = t_vec - t_vec_par
+    t_vec_perp = t_vec_perp/np.linalg.norm(t_vec_perp)
+
+    # Compute vector othogonal to both rotation axis and orthogonal
+    # part of thickness vector
+    orth = np.cross(rot_axis, t_vec_perp)
+    orth = orth/np.linalg.norm(orth)
+
+    # Convert angle from degress to radians
+    rot_ang_norm = np.deg2rad(rot_ang_norm)
+    
+    # Determine part of rotated vector parallel to original
+    rot_par = np.cos(rot_ang_norm)
+    # Determine part of rotated vector orthogonal to original
+    rot_perp = np.sin(rot_ang_norm)
+
+    # Compute orthogonal part of thickness vector after rotation
+    t_vec_perp_rot = rot_par*t_vec_perp + rot_perp*orth
+    # Compute thickness vector after rotation
+    t_vec_rot = t_vec_perp_rot + t_vec_par
+    t_vec_rot = t_vec_rot/np.linalg.norm(t_vec_rot)
+
+    # Orient cross-section in its plane such that it faces the global origin
+
+    # Extract initial path point
+    pos = cubit.vertex(path_origin).coordinates()
+
+    # Project position vector onto cross-section
+    pos_proj = pos - (np.inner(pos, norm)/np.inner(pos, pos))*norm
+    pos_proj = pos_proj/np.linalg.norm(pos_proj)
```

Comment:
If we have a `unit_vector` function this could become:
```suggestion
    pos_proj = unit_vector(pos - np.inner(unit_vector(pos),norm) * norm)
```

Reply:
`np.inner(pos,pos) = magnitude(pos)**2`, so unless I'm missing something, I'm not sure this change would produce the same result

Reply:
Aaah - you're right, but now I wonder about why this is correct?  It seems like an unusual vector operation?

Reply:
The equation projects the position vector `pos` onto the cross-section plane (after the cross-section is oriented such that it's normal is aligned with the filament tangent). Essentially what it is doing is taking `pos` and subtracting from it the projection of `pos` onto `norm` (the cross-section normal, also the filament tangent), which is the component of `pos` that is normal to the cross-section. Thus, we are left with the component of `pos` that is parallel to the cross-section